### PR TITLE
Dx 723 describe description field

### PR DESCRIPTION
**Pull request by marinafrydenb**
this is just a suggestion, please help me to improve. and i have no clue how big the pictures should be to look pretty.

The field is not included other places yet.

### CODE REVIEW

File: _includes/field-description.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+```
+
+![description field in redirect view][description-field-redirect]{:width="1200px"
+:height="500px"}
+
+![description field in seamless-view][description-field-seamless]{:width="1200px"
+:height="500px"}
+
+[description-field-redirect]:/assets/screenshots/description-field/description-field-redirect.png
```
Comment (NullableInt) [inline _includes/field-description.md]:
```suggestion
[description-field-redirect]: /assets/screenshots/description-field/description-field-redirect.png
```
For consistency.

File: _includes/field-description.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+## Description
+
+The `Description` field is a textual definition max 40 characters of the
```
Comment (asbjornu) [inline _includes/field-description.md]:
```suggestion
The `description` field is a textual definition max 40 characters of the
```

File: _includes/field-description.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+purchase. It is needed to specify what the purchase actually is for
+the consumer. Below you will find a abbreviated code example of Card Payments
+`Purchase`.
+As you can see the `description` field is set to be
+`"test purchase - orderNumber28749347"` in the the code example.
+The images below will show you the payment UI for the
+Redirect and Seamless View scenario. Notice that for Redirect, the description
+will be shown as `"test purchase - orderNumber28749347"`, as set in the code example.
+For the Seamless view scenario, the description is not shown in the payment
+window and therefore it doesnt need to be included.
```
Comment (asbjornu) [inline _includes/field-description.md]: I think `description` is required for all requests, so it must be included regardless. I think that's why people wonder why it's required.

```suggestion
window, but it is still required in the initial request.
```

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 13
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5258688&bid=17228984)

File: _includes/field-description-description.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{%- capture description -%}
+    A 40 character length textual [description][description] of the purchase.
```
Comment (asbjornu) [inline _includes/field-description-description.md:2]: We need the footnote link for `[description]` in the include and probably a `payment_instrument` parameter so the link can be constructed relative to the payment instrument.

File: _includes/card-purchase.md [hunk 5]
```diff
@@ -279,6 +279,7 @@ Content-Type: application/json
 | └─➔&nbsp;`href`                   | `string`     | The target URI to perform the operation against.                                                                                                                                                                                                                                                                                                           |
 | └─➔&nbsp;`rel`                    | `string`     | The name of the relation the operation has to the current resource.                                                                                                                                                                                                                                                                                        |
 
+[description]: payments/card/other-features#description
```
Comment (asbjornu) [inline _includes/card-purchase.md:282]: This footnote link should go into the include and isn't needed here, I think.

File: _includes/payment-order-purchase.md [hunk 6]
```diff
@@ -105,7 +105,7 @@ Content-Type: application/json
...
-|  ✔︎︎︎︎︎  | └➔&nbsp;`description`             | `string`     | The description of the payment order.                                                                                                                                                                                                                                                                                                                                                                         |
+|  ✔︎︎︎︎︎  | └➔&nbsp;`description`             | `string`     | {% include field-description-description.md %}                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (asbjornu) [inline _includes/payment-order-purchase.md:108]:
```suggestion
|  ✔︎︎︎︎︎  | └➔&nbsp;`description`             | `string(40)` | {% include field-description-description.md %}                                                                                                                                                                                                                                                                                                                                                                       |
```

File: payments/vipps/seamless-view.md [hunk 7]
```diff
@@ -310,7 +310,7 @@ Content-Type: application/json
...
 | └➔&nbsp;`amount`             | `integer` | {% include field-description-amount.md currency="NOK" %}                                                                                                                                                     |
 | └➔&nbsp;`vatAmount`          | `integer` | {% include field-description-vatamount.md currency="NOK" %}                                                                                                                                                  |
-| └➔&nbsp;`description`        | `string`  | A human readable description of maximum 40 characters of the transaction.                                                                                                                                    |
+| └➔&nbsp;`description`        | `string`  | {% include field-description-description.md %}                                                                                                                                    |
```
Comment (asbjornu) [inline payments/vipps/seamless-view.md:313]:
```suggestion
| └➔&nbsp;`description`        | `string(40)` | {% include field-description-description.md %}                                                                                                                                    |
```

File: payments/vipps/seamless-view.md [hunk 8]
```diff
@@ -389,6 +389,7 @@ sequenceDiagram
 [callbackurl]: /payments/vipps/other-features#callback
 [cancellations]: /payments/vipps/after-payment#cancellations
 [captures]: /payments/vipps/after-payment#captures
+[description]: /payments/vipps/other-features#description
```
Comment (asbjornu) [inline payments/vipps/seamless-view.md:392]: I don't think this footnote link is needed.

```suggestion
```

File: payments/vipps/redirect.md [hunk 9]
```diff
@@ -416,6 +416,7 @@ purchased products) have to make a [Capture][capture] or
                          next_href="seamless-view"
                          next_title="Next: Implement Seamless view" %}
 
+[description]: /payments/vipps/other-features#description
```
Comment (asbjornu) [inline payments/vipps/redirect.md:419]:
```suggestion
```

File: payments/swish/seamless-view.md [hunk 10]
```diff
@@ -330,6 +330,7 @@ next_href="after-payment" next_title="Next: After Payment" %}
 
 [callback]: /payments/swish/other-features#callback
 [create-payment]: /payments/swish/after-payment#create-payment
+[description]: /payments/swish/other-features#description
```
Comment (asbjornu) [inline payments/swish/seamless-view.md:333]:
```suggestion
```

File: payments/swish/redirect.md [hunk 11]
```diff
@@ -243,6 +243,7 @@ sequenceDiagram
 {% include iterator.html prev_href="direct" prev_title="Back: Direct"
 next_href="seamless-view" next_title="Next: Seamless View" %}
 
+[description]: payments/swish/other-features#description
```
Comment (asbjornu) [inline payments/swish/redirect.md:246]:
```suggestion
```

File: payments/swish/direct.md [hunk 12]
```diff
@@ -392,6 +392,7 @@ sequenceDiagram
 {% include iterator.html prev_href="introduction" prev_title="Back: Introduction"
 next_href="redirect" next_title="Next: Redirect" %}
 
+[description]: /payments/swish/other-features#description
```
Comment (asbjornu) [inline payments/swish/direct.md:395]:
```suggestion
```

File: payments/mobile-pay/redirect.md [hunk 13]
```diff
@@ -379,6 +379,7 @@ sequenceDiagram
                          next_href="after-payment"
                          next_title="Next: After Payment" %}
 
+[description]: /payments/mobile-pay/other-features#description
```
Comment (asbjornu) [inline payments/mobile-pay/redirect.md:382]:
```suggestion
```

File: payments/invoice/seamless-view.md [hunk 14]
```diff
@@ -305,6 +305,7 @@ next_href="direct" next_title="Next: Direct" %}
 [cancel]: /payments/invoice/after-payment#cancellations
 [capture]: /payments/invoice/after-payment#captures
 [create-payment]: /payments/invoice/other-features#create-payment
+[description]: /payments/invoice/other-features#description
```
Comment (asbjornu) [inline payments/invoice/seamless-view.md:308]:
```suggestion
```

File: payments/invoice/redirect.md [hunk 15]
```diff
@@ -255,6 +255,7 @@ next_href="seamless-view" next_title="Next: Seamless View" %}
 [callback]: /payments/invoice/other-features#callback
 [cancel]: /payments/invoice/after-payment#cancellations
 [capture]: /payments/invoice/after-payment#captures
+[description]: /payments/invoice/other-features#description
```
Comment (asbjornu) [inline payments/invoice/redirect.md:258]:
```suggestion
```

File: payments/invoice/direct.md [hunk 16]
```diff
@@ -403,6 +403,7 @@ next_href="capture" next_title="Next: Capture" %}
 [callback]: /payments/invoice/other-features#callback
 [cancel]: /payments/invoice/after-payment#cancellations
 [capture]: /payments/invoice/capture
+[description]: payments/invoice/other-features#description
```
Comment (asbjornu) [inline payments/invoice/direct.md:406]:
```suggestion
```

File: _includes/payment-order-purchase.md [hunk 17]
```diff
@@ -147,3 +147,5 @@ Content-Type: application/json
 |  ✔︎︎︎︎︎  | └─➔&nbsp;`amount`                 | `integer`    | {% include field-description-amount.md %}                                                                                                                                                                                                                                                                                                                                                                     |
 |  ✔︎︎︎︎︎  | └─➔&nbsp;`vatAmount`              | `integer`    | {% include field-description-vatamount.md %}                                                                                                                                                                                                                                                                                                                                                                  |
 {% include risk-indicator-table.md %}
+
+[description]: /checkout/other-features#description
```
Comment (asbjornu) [inline _includes/payment-order-purchase.md:151]:
```suggestion
```

File: _includes/description.md [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+purchase. It is needed to specify what the purchase actually is for
+the consumer. Below you will find a abbreviated code example of Card Payments
+`Purchase`.
+As you can see the `description` field is set to be
+`"test purchase - orderNumber28749347"` in the the code example.
+The images below will show you the payment UI for the
+Redirect and Seamless View scenario. Notice that for Redirect, the description
+will be shown as `"test purchase - orderNumber28749347"`, as set in the code example.
+For the Seamless view scenario, the description is not shown in the payment
+window, but it is still required in the initial request.
```
Comment (asbjornu) [inline _includes/description.md:13]:
```suggestion
The `description` field is a 40 character length textual summary of the
purchase. It is needed to specify what payer is actually purchasing. Below you
will find an abbreviated Card Payments `Purchase` request.

As you can see the `description` field is set to be `test purchase -
orderNumber28749347` in the the code example. The images below will show you the
payment UI for the Redirect and Seamless View scenario.

{% include alert.html type="neutral" icon="info" body="Notice that for Redirect,
the description will be shown as `test purchase - orderNumber28749347`, as set
in the code example. For the Seamless View scenario, the description is not
shown in the payment window, but it is still required in the initial request."
%}
```

File: payments/swish/other-features.md [hunk 19]
```diff
@@ -172,6 +172,8 @@ be connected to a Swish account.
 
 {% include payment-link.md show_3d_secure=false show_authorization=false %}
 
+{% include description.md %}
```
Comment (asbjornu) [inline payments/swish/other-features.md:175]: Isn't a "Description" header needed here?

File: _includes/description.md [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+the consumer. Below you will find a abbreviated code example of Card Payments
+`Purchase`.
+As you can see the `description` field is set to be
+`"test purchase - orderNumber28749347"` in the the code example.
+The images below will show you the payment UI for the
+Redirect and Seamless View scenario.
+{% include alert.html type="neutral" icon="info" body="Notice that for Redirect,
+the description will be shown as `test purchase - orderNumber28749347`, as set
+in the code example. For the Seamless View scenario, the description is not
+shown in the payment window, but it is still required in the initial request." %}
```
[the remark this answers is not in the document]
Comment (asbjornu) [reply _includes/description.md:12]: @marinafrydenb, did you not see this? 😃 ☝️

File: _includes/field-description-description.md [hunk 21]
```diff
@@ -0,0 +1,15 @@
+{% assign payment_instrument = include.payment_instrument | default: "creditcard" %}
+{%- capture description -%}
+    A 40 character length textual [description](/payments/{{ payment_instrument }}/other-features#description) of the purchase.
```
[the remark this answers is not in the document]
Comment (asbjornu) [reply _includes/field-description-description.md:8]: This still looks like it's unresolved?

File: _includes/transaction-response.md [hunk 22]
```diff
@@ -100,7 +100,7 @@ Content-Type: application/json
 | └─➔&nbsp;`number`                 | `string`  | The transaction `number`, useful when there's need to reference the transaction in human communication. Not usable for programmatic identification of the transaction, for that `id` should be used instead. |
 | └─➔&nbsp;`amount`                 | `integer` | {% include field-description-amount.md %}                                                                                                                                                                    |
 | └─➔&nbsp;`vatAmount`              | `integer` | {% include field-description-vatamount.md %}                                                           |
-| └─➔&nbsp;`description`            | `string`  | A human readable description of maximum 40 characters of the transaction.                                                                                                                                    |
+| └─➔&nbsp;`description`            | `string`  | {% include field-description-description.md payment_instrument=payment_instrument %}                                                                                                                                    |
```
Comment (asbjornu) [inline _includes/transaction-response.md:103]: In this include, the variable name is called `instrument`, not `payment_instrument`. I suggest fixing it to be `payment_instrument` everywhere.

File: payments/swish/direct.md [hunk 23]
```diff
@@ -114,7 +114,7 @@ Content-Type: application/json
...
 | {% icon check %} | └─➔&nbsp;`vatAmount`         | `integer`     | {% include field-description-vatamount.md %}                                                                                                                                                                                                                                                       |  |
-| {% icon check %} | └➔&nbsp;`description`        | `string(40)`  | A textual description max 40 characters of the purchase.                                                                                                                                                                                                                                           |
+| {% icon check %} | └➔&nbsp;`description`        | `string(40)`  | {% include field-description-description.md payment_instrument ="swish" %}                                                                                                                                                                                                                                           |
```
Comment (asbjornu) [inline payments/swish/direct.md:117]:
```suggestion
| {% icon check %} | └➔&nbsp;`description`        | `string(40)`  | {% include field-description-description.md payment_instrument="swish" %}                                                                                                                                                                                                                                           |
```

File: payments/swish/redirect.md [hunk 24]
```diff
@@ -123,7 +123,7 @@ Content-Type: application/json
...
 | {% icon check %} | └─➔&nbsp;`vatAmount`         | `integer`     | {% include field-description-vatamount.md %}                                                                                                                                                                                                                                                       |
-| {% icon check %} | └➔&nbsp;`description`        | `string(40)`  | A textual description max 40 characters of the purchase.                                                                                                                                                                                                                                           |
+| {% icon check %} | └➔&nbsp;`description`        | `string(40)`  | {% include field-description-description.md payment_instrument ="swish" %}                                                                                                                                                                                                                                           |
```
Comment (asbjornu) [inline payments/swish/redirect.md:126]:
```suggestion
| {% icon check %} | └➔&nbsp;`description`        | `string(40)`  | {% include field-description-description.md payment_instrument="swish" %}                                                                                                                                                                                                                                           |
```

File: _includes/transaction-response.md [hunk 25]
```diff
@@ -1,4 +1,4 @@
-{% assign instrument = include.payment_instrument | default: "paymentorder" %}
+{% assign payment_instrument = include.payment_instrument | default: "paymentorder" %}
```
Comment (NullableInt) [inline _includes/transaction-response.md:1]: You also have to update the rest of this file to use `{{ payment_instrument }}` and not `{{ instrument }}`

File: payments/invoice/direct.md [hunk 26]
```diff
@@ -225,27 +225,27 @@ Content-Type: application/json
...
+| └─➔&nbsp;`method`        | `string`     | The HTTP method to use when performing the operation.                                                                                                                                                                                                                                                                                                      |
+| └─➔&nbsp;`href`          | `string`     | The target URI to perform the operation against.                                                                                                                                                                                                                                                                                                           |
+| └─➔&nbsp;`rel`           | `string`     | The name of the relation the
+operation has to the current resource.|
```
Comment (NullableInt) [inline payments/invoice/direct.md:248]:
```suggestion
| └─➔&nbsp;`rel`           | `string`     | The name of the relation the operation has to the current resource.|
```

File: payments/mobile-pay/other-features.md [hunk 27]
```diff
@@ -201,6 +201,8 @@ expanded sub-resources inlined.
 }
 ```
 
+{% include description.md %}
```
Comment (NullableInt) [inline payments/mobile-pay/other-features.md:204]: Why is this included twice on this page?

**Comment by NullableInt** [comment]
@marinafrydenb please resolve all comments before you re-request a review.

File: _includes/transaction-response.md (hunk 25 above)
Comment (NullableInt) [inline _includes/transaction-response.md:1]: You still have to update the rest of this file.

File: _includes/description.md [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+
+As you can see the `description` field is set to be
+`"test purchase - orderNumber28749347"` in the the code example.
+The images below will show you the payment UI for the
+Redirect and Seamless View scenario.
+
+{% include alert.html type="neutral" icon="info" body="Notice that for Redirect,
+the description will be shown as `test purchase - orderNumber28749347`, as set
+in the code example. For the Seamless View scenario, the description is not
+shown in the payment window, but it is still required in the initial request." %}
```
Comment (asbjornu) [inline _includes/description.md:14]: This review suggestion seems to have been ignored.

```suggestion
The `description` field is a 40 character length textual summary of the
purchase. It is needed to specify what payer is actually purchasing. Below you
will find an abbreviated Card Payments `Purchase` request.

As you can see the `description` field is set to be `test purchase -
orderNumber28749347` in the the code example. The images below will show you the
payment UI for the Redirect and Seamless View scenario.

{% include alert.html type="neutral" icon="info" body="Notice that for Redirect,
the description will be shown as `test purchase - orderNumber28749347`, as set
in the code example. For the Seamless View scenario, the description is not
shown in the payment window, but it is still required in the initial request."
%}
```
